src/client/item_visuals_manager.h

```diff
@@ -46,9 +57,25 @@ struct ItemVisualsManager
 	struct ItemVisuals
 	{
 		video::ITexture *inventory_texture;
-		ItemMesh wield_mesh;
+		video::ITexture *inventory_overlay_texture;
+		ItemMesh item_mesh;
 		Palette *palette;
 
+		// ItemVisuals owns the frames vector of the inventory and overlay image,
+		// and stores an AnimationInfo to draw animated non-mesh items,
+		struct OwnedAnimationInfo {
```

Comment:
```suggestion
		// OwnedAnimationInfo exists to own the frames vector referenced by
		// the contained AnimationInfo.
		struct OwnedAnimationInfo {
```

if I understand correctly?
(might need rewording to fit the existing comment)